### DISCUSSION TITLE
LDAP PHP Allowed memory size exhausted

### DISCUSSION THREAD

**Discussion by twisted3motions** [snipe/snipe-it#17656] [Q&A]
#### Please confirm you have done the following before posting your bug report:

- [X ] I have enabled debug mode 
- [X ] I have read [checked the Common Issues page](https://snipe-it.readme.io/docs/common-issues)

**Describe the bug**
Testing LDAP results in a long hang and then  "500 Server Error. Please check your server logs for more information."

**Expected behavior**
A box with identical LDAP field settings but Ubuntu 18.04 LTS, Apache, and SnipeIT 4.9.4 works fine.  Instant connection.  

**Server (please complete the following information):**
No go on box with these.

 - Snipe-IT Version 5.0.6
 - OS: Ubuntu 20.04 LTS
 - Web Server: Apache
 - PHP Version: 7.4

**Error Messages**

Debug doesn't really show anything but this:  WARNING!: This application is running in production mode with debugging enabled. This can expose sensitive data if your application is accessible to the outside world. Disable debug mode by setting the APP_DEBUG value in your .env file to false.

However the apache logs tell me this:

[Fri Nov 13 21:31:38.765170 2020] [php7:error] [pid 2572] [client IPADDRESS:53200] PHP Fatal error:  Allowed memory size of 1073741824 bytes exhausted (tried to allocate 32768 bytes) in /var/www/snipeit/vendor/symfony/debug/Exception/FatalErrorException.php on line 1, referer: http://snipeitaddress/admin/ldap

Boosted PHP.ini was initially 128MB like on the old server.  Boosted to 1GB, and it just hung longer before spitting out same general error.  Do I need to boost this even further?

**Comment by uberbrady** [comment]
You'd have to have a *lot* of users in your directory to need more than a gig. Try instead adjusting the `LDAP_MEM_LIM` `.env` var instead - we actually do some overrides in there specifically just on the sync call, and your php.ini settings might be getting overwritten by our smaller numbers.

**Comment by samotelf** [comment]
@twisted3motions in much older versions of Snipe I had to change the memory_limit=256 in /etc/php.ini
Hope it helps.
